File: scala/src/main/scala/com/authdog/exceptions/AuthenticationException.scala
package com.authdog.exceptions

/**
 * Exception thrown when authentication fails
 */
class AuthenticationException(message: String = "Authentication failed", cause: Option[Throwable] = None) 
  extends AuthdogException(message, cause)

object AuthenticationException {
  def apply(message: String): AuthenticationException = new AuthenticationException(message)
  def apply(message: String, cause: Throwable): AuthenticationException = new AuthenticationException(message, Some(cause))
}

File: java/src/main/java/com/authdog/exceptions/AuthenticationException.java
package com.authdog.exceptions;

/**
 * Exception thrown when authentication fails.
 */
public class AuthenticationException extends AuthdogException {

    /**
     * Constructor with message.
     * @param messageParam Error message
     */
    public AuthenticationException(final String messageParam) {
        super(messageParam);
    }

    /**
     * Constructor with message and cause.
     * @param messageParam Error message
     * @param causeParam Cause of the exception
     */
    public AuthenticationException(final String messageParam,
                                   final Throwable causeParam) {
        super(messageParam, causeParam);
    }
}
